enhancements/microshift/microshift-default-cni.md

```diff
@@ -0,0 +1,290 @@
+plugin which would provide Pod to Pod communication in a single-node environment
+and nothing else.
+
+A very important characteristic necessary for any CNI plugin used in MicroShift
+is a low-enough RAM, Disk, CPU footprint and boot times, as explained in the summary,
+the target platforms for MicroShift are small SBCs or SoCs where most of
+the computing capability must remain available for the final applications,
+otherwise MicroShift stops being useful in it's target environment.
+
+### User Stories
+
+An application developer builds applications where:
+1. Pods need to talk to each other via TCP o UDP connectivity.
+2. Pods need to talk to services on the LAN or Internet via TCP or UDP.
+3. Some pods need be exposed as NodePort types of services on the LAN.
+4. Some pods need to be exposed to other pods as ClusterIP services types.
+
+An application developer builds applications where some pods must
+have limited connectivity to other pods, or to the external network (i.e.
+plugin services from vendors which must only contact specific internal APIs)
+
+The edge device with MicroShift and the final application is deployed on an
+IPv6, or DualStack IPV4/IPV6 network.
+
+As a device owner I can protect the edge device with a host-level firewall
+(firewalld, iptables, nftables), and that type of configuration is compatible
+with the CNI.
+
+The edge device should be capable of deploying on a network with dynamic address
+provisioning like DHCP, SLAAC, or DHCPv6 and remain functional when the IP address
+changes.
+
+### Goals
+ 
+* Implement a default CNI that satisfies the described user histories while
+  minimizing the impact on footprint.
+
+* The CNI plugin should run without the ClusterNetworkOperator, providing
+  the `microshift-networking` elements any basic system configuration elements,
+  tools or dependencies to make the CNI plugin work.
+
+* Follow the [MicroShift design
+  principles](https://github.com/openshift/microshift/blob/main/docs/design.md)
+
+### Non-Goals
+
+* We do not plan to support multi-node clusters.
+* We are not trying to address high availability within the CNI, 
+  high availability should be acomplished by having multiple edge devices working
+  as single node MicroShifts.
+
+* LoadBalancer type of Services, as this is out of the scope of the CNI.
+
+## Proposal
+
+To meet the requirements described in the user histories, as well footprint
+requirements of the edge environment, an optimized version of OVN Kubernetes
+is proposed.
+
+This option aligns better with Red Hat's strategy on networking and avoids
+the need to maintain bridge-cni which is deprecated in podman 4.
+
+Using OVN-kubernetes is only possible thanks to the following optimizations:
+* Single node operation
+* Communication via unix sockets, avoiding the need of TLS & Certificate creation
+  at boot time
+* No leader election, as there is no high availability or multi-node operation.
```

Comment:
There are two leader elections, one for ovn database, the other for ovnk-master. we only removed the leader election for ovndb as of now, could make it explicit that no leader election for ovn db.

Reply:
Do we need leader election in the ovnk-master for anything?, can we get rid of it? (I guess if you didn't already is because the cost/benefit relationship isn't good)

Reply:
we don't need leader election for ovnk-master. yes it can be removed in theory.
I didn't touch it because the benefit we'd get may not be worth the time when doing the early evaluation.